packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -137,7 +137,9 @@ export class TableLayout<T> extends ListLayout<T> {
     let width = 0;
     for (let i = colIndex; i < colIndex + colspan; i++) {
       let column = this.collection.columns[i];
-      width += this.getColumnWidth(column.key);
+      if (column?.key) {
+        width += this.getColumnWidth(column.key);
```

Comment:
this is treating a not found key as 0 essentially, that could result in one column overlapping another during animation
talked with devon for a while about this, we don't have a great solution yet, but wanted to get it noted on the review
we thought maybe caching the previous collection might work, but if two columns are removed rapidly, we'd be back in the same situation
another possible option is a WeakMap of column widths associated with nodes

Reply:
While testing this I did see some overlap, but only when adding and removing columns very quickly.